src/main/java/org/isf/lab/service/LabIoOperations.java

```diff
@@ -76,6 +79,27 @@ public List<LaboratoryRow> getLabRow(Integer code) throws OHServiceException {
 		return rowRepository.findByLaboratory_Code(code);
 	}
 
+	/**
+	 * Return the list of exams ({@link Laboratory}s) divided by pages.
+	 * 
+	 * @param oneWeek
+	 * @param pageNo
+	 * @param pageSize
+	 * @return the list of {@link Laboratory}s (could be empty)
+	 * @throws OHServiceException
+	 */
+	public List<Laboratory> getLaboratory(boolean onWeek, int pageNo, int pageSize) throws OHServiceException {
+		LocalDateTime time2 = TimeTools.getDateToday24();
+		LocalDateTime time1 = time2.minusWeeks(1);
+		Pageable pageable = PageRequest.of(pageNo, pageSize);
+		if (onWeek) {
```

Comment:
Why aren't the 2 time calculations inside the `if` block? That is the only place `time2` and `time1` are used.